File: programmers/Level 1/체육복.py
def solution(n, lost, reserve):
    answer = 0
    students = [1 for _ in range(n)] # 초기화

    for idx in lost:
        students[idx - 1] -= 1

    for idx in reserve:
        students[idx - 1] += 1

    for i in range(n):
        if students[i]: # 체육복이 있는 경우 continue
            continue
        elif i > 0 and students[i - 1] > 1: # 이전 번호 확인
            students[i] += 1
            students[i - 1] -= 1
        elif i < n - 1 and students[i + 1] > 1: # 다음 번호 확인
            students[i] += 1
            students[i + 1] -= 1

    answer = len([i for i in students if i])

    return answer

File: programmers/Level 1/체육복.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(int n, vector<int> lost, vector<int> reserve) {
    int answer = 0;

    map<int, int> students; // 학생 별 체육복 수 저장
    for(int i = 1; i <= n; i++)
        students[i] = 1;
    for (int l: lost)
        students[l] -= 1;
    for (int r: reserve)
        students[r] += 1;

    // 앞번호 학생의 체육복 빌리기 -> 뒷번호 학생의 체육복 빌리기
    for (int i = 1; i <= n; i++){
        if(students[i] == 0){
            if (i != 1 && students[i - 1] > 1){
                students[i] += 1;
                students[i - 1] -= 1;
            }
            else if (i != n && students[i + 1] > 1){
                students[i] += 1;
                students[i + 1] -= 1;
            }
        }
    }

    // 체육복 없는 학생의 수
    for (auto a: students){
        if(a.second != 0)
            answer++;
    }
    return answer;
}
